atc/db/prototype.go

```diff
@@ -231,6 +237,10 @@ func (p *prototype) CreateBuild(ctx context.Context, manuallyTriggered bool, pla
 	return build, true, nil
 }
 
+func (p *prototype) CreateInMemoryBuild(context.Context, atc.Plan, util.SequenceGenerator) (Build, error) {
+	panic("not implemented")
```

Comment:
Because panic will explicitly tell `prototype` not supporting in-memory build yet. Returning nil might lead to unpredictable errors.

Reply:
I think panic explicitly tells that prototype doesn't support in-memory build, so that future code changer won't try to call it.

Reply:
Sorry to jump here uninvited :-), but I do share clarafu's opinion on trying to avoid panics at all costs.
In this particular case, it seems to me that the same intent "prototype doesn't support in-memory build" can be obtained by returning an error, which is probably also what clarafu meant (given the signature)

```go
func (p *prototype) CreateInMemoryBuild(context.Context, atc.Plan, util.SequenceGenerator) (Build, error) {
	return nil, fmt.Errorf("not implemented")
```

> Returning nil might lead to unpredictable errors.

Given the signature of the return type, this will not happen, as long as the caller does its duty and checks for errors...

Or am I missing something?

Reply:
Yeah, as I mentioned in the other comment, I can change to use error to replace panic.